CAIPs/caip-25.md

```diff
@@ -93,20 +123,20 @@ Example:
 ```
 
 The JSON-RPC method is labelled as `provider_authorization` and both the
-"requiredNamespaces" and "optionalNamespaces" arrays are populated with 
-`namespace` objects each named after the scope of authorization:
-1. EITHER an entire ChainAgnostic [namespace][] 
-2. OR a specific [CAIP-2][] in that namespace.
+"requiredScopes" and "optionalScopes" arrays are populated with 
+"scope objects" each named after the scope of authorization requested:
+1. EITHER an entire [CAIP-104][] [namespace][]
+2. OR a specific [CAIP-2][]-identified chain in a specific namespace.
 
-Each `namespace` object contains the following parameters:
+Each scope object contains the following parameters:
 - chains - array of [CAIP-2][]-compliant `chainId`'s. This parameter MAY be
   omitted if a single-chain scope is already declared in the index of the object.
 - methods - array of JSON-RPC methods expected to be used during the session
```

Comment:
As discussed in today's editorial call: maybe using method names is not completely covering it - there is different behavior in methods between different clients (@TimDaub surfaced this)
My idea to attack it would be to create a registry (like we have namespaces) of possible methods - maybe defining them like here: https://github.com/ethereum/execution-apis/blob/main/src/eth/block.yaml - so we could cover e.g. the case of having one method_name with different parameters

Reply:
I agree that discrepancies in APIs is an issue worth attacking-- my counterargument, though, is that CAIP shouldn't host or run the registry that attacks it! I think the way 211 is written, it presumes that, for example, if geth or erigon use slightly different versions of a method, they can each publish `rpcDocuments` somewhere canonical* and a dapp can request the the geth RPC doc be prioritized over the execution-apis RPC doc, or a list of Geth clients be requested as `rpcEndpoints`. the fact that they're ordered arrays allows the wallet a lot of freedom in how it replies to that request, beyond just y/n.

* = that somewhere canonical is probably for the EF to decide, right? it could be an eip155-wide registry or domain controlled by a central party, or it could just be https://github.com/ethereum/go-ethereum ; I tried to write 211 such that RPC docs are just URLs, and trust decisions about categorizing or sorting those can happen based on origins, longevity, etc.

Reply:
Actually I disagree with this as clients should not be changing behavior for methods without renaming them

If a client behaves differently for example with `eth_call` then the clients need to coordinate in a standard spec or the method is renamed

adding a client suffix is only going to incentivize more fragmentation and less collaboration

if clients created this problem then they should fix it because they are mis-implementing methods

can you imagine if every wallet implemented `eth_sendTransaction` differently??

Reply:
I agree it is not ideal - but it is reality and something we need to deal with unfortunately. What is your counter-proposal to deal with the situation?

Reply:
At least with this solution we can detect with a handshake that something would fail and maybe can fallback or handle it gracefully instead of having weird unpredictable errors.

Reply:
the way I see it, there does not need to be a registry of multiple RPC documents-- there is the one RPC document defined by `execution-api`s, which namespaces/eip155/caip211 can define as the "default" definition of any methods authorized in eip155 scope objects by CAIP-25 negotiations... and dapps and wallets are free to agree to use additional documents instead of or in addition to the namespace-wide default.  

Can you make the RPC WG today to discuss further, Ligi?  We can always invite Shane back to discuss how he/the openRPC team sees it

Reply:
Merging this PR as-is, but this entire thread is still very pertinent to #211 which lays out the RPC document stuff (in a way that can be profiled for each namespace!), so let's continue the thread there!